src/test/javascript/spec/component/complaints/complaints.spec.ts

```diff
@@ -0,0 +1,60 @@
+import { ComponentFixture, TestBed, async, tick, fakeAsync } from '@angular/core/testing';
+import * as chai from 'chai';
+import { By, BrowserModule } from '@angular/platform-browser';
+import { ComplaintService } from 'app/entities/complaint/complaint.service';
+import { MockComplaintResponse, MockComplaintService } from '../../mocks/mock-complaint.service';
+import { MockAlertService } from '../../helpers/mock-alert.service';
+import { JhiAlertService } from 'ng-jhipster';
+import { ArTEMiSSharedModule } from 'app/shared';
+import { MomentModule } from 'ngx-moment';
+import { ClipboardModule } from 'ngx-clipboard';
+import { DebugElement } from '@angular/core';
+import { HttpClientModule } from '@angular/common/http';
+import { ComplaintsComponent } from 'app/complaints';
+
+const expect = chai.expect;
+describe('ComplaintsComponent', () => {
+    let comp: ComplaintsComponent;
+    let fixture: ComponentFixture<ComplaintsComponent>;
+    let debugElement: DebugElement;
+
+    beforeEach(async(() => {
+        TestBed.configureTestingModule({
+            imports: [BrowserModule, ArTEMiSSharedModule, MomentModule, ClipboardModule, HttpClientModule],
+            declarations: [ComplaintsComponent],
+            providers: [
+                {
+                    provide: ComplaintService,
+                    useClass: MockComplaintService,
+                },
+                {
+                    provide: JhiAlertService,
+                    useClass: MockAlertService,
+                },
+            ],
+        })
+            .compileComponents()
+            .then(() => {
+                fixture = TestBed.createComponent(ComplaintsComponent);
+                comp = fixture.componentInstance;
+                debugElement = fixture.debugElement;
+            });
+    }));
+
+    it('should initialize with correct values for complaints service', fakeAsync(() => {
```

Comment:
Cool that you wrote tests!

Besides the initialization, could you please also write a couple of tests that verifies the new functionality introduced by you? 
It's no must to create tests that are not related to your changes, but we should watch out that we don't decrease the test coverage of the project.

Reply:
I will add couple more tests, after all other parts are done

Reply:
Added some more tests